### DISCUSSION TITLE
One button to click all active buttons

### DISCUSSION THREAD

**Discussion by gmentzik** [Budibase/budibase#8914] [Help]
I have a page with multiple forms in a repeater. Each form has a submit button that is enabled if field is altered compared to initial value.
See below:
![image](https://user-images.githubusercontent.com/65217733/205341828-af9580bf-a2bd-495e-931e-e8a665692f7d.png)

I wonder if the a way to create a button that will be able to click all active buttons of those forms at once.
The data-name attribute of that button is "UpdateButton".
Any suggestions?

**Comment by melohagan** [comment]
Hey @gmentzik 

As far as I'm aware this isn't directly possible. 

What you may be able to do is save each form/row state to a JS array in [App state](https://docs.budibase.com/docs/app-state) using the **On Change** event on each of your form fields.

Then your **Save All** button could send that App state array to an automation that saves the rows on [loop](https://docs.budibase.com/docs/looping) through the [App action](https://docs.budibase.com/docs/app-action) trigger.

**Reply by gmentzik** [reply]
@melohagan 
Thank you for the response. 

I have already tried the solution (case multiple forms , one in each repeater block), to store each field in a Json object  in a state and then use automation loop through array, based on the suggestion you had given to another user, but I faced 2 big issues and dropped it as solution. 
1) Response of browser (chrome) during state updates is slow and the more complex the form the worse it gets, especially if things need to be updated in repeater like having text fields and multiple checkboxes inside.
2) Automation looping does not report errors. So if one row has gone bad or even nothing is saved it happily returns success! There is an error handling mechanism missing there. I would like to have the options to make save row loop automation a) transactional (revert in case of error)  and inform, b) stop in case of error and inform, c) skip error but show warning/info message to user for failing rows.

For simple forms (eg. only 1/2 text fields per form), from issues mentioned above (1) is acceptable but (2) is a serious problem for me, unless I am missing something here...

**Reply by melohagan** [reply]
The automations are async i.e.g fire and forget, so at the moment there is no feedback mechanism. The best you can do is use [Form validation](https://docs.budibase.com/docs/form-validation) to try and prevent bad requests being sent to the automation in the first place.

**Reply by gmentzik** [reply]
> The automations are async i.e.g fire and forget, so at the moment there is no feedback mechanism. The best you can do is use [Form validation](https://docs.budibase.com/docs/form-validation) to try and prevent bad requests being sent to the automation in the first place.

The problem if that if in looping save row completely fails there is no way to know about this unless you check the database results, also even if you perform your best doing form validation in case the Javascript goes wrong and array is missing info, bad schema etc, still you have no way to know about it. The best guess is to create some custom JS validator before triggering automation.
